internal/users/tempentries/preauth_test.go

```diff
@@ -135,12 +200,13 @@ func TestPreAuthUserByIDAndName(t *testing.T) {
 	}
 }
 
-func checkPreAuthUser(t *testing.T, user types.UserEntry) {
+func checkPreAuthUser(t *testing.T, user types.UserEntry, options ...golden.Option) {
 	t.Helper()
 
 	// The name field is randomly generated, so unset it before comparing the user with the golden file.
-	require.NotEmpty(t, user.Name, "Name should not be empty")
-	user.Name = ""
+	require.True(t, strings.HasPrefix(user.Name, UserPrefix),
+		"Name should have %q prefix", UserPrefix)
+	user.Name = UserPrefix + "-XXXXXXXX"
```

Comment:
We can use the same pattern we already use in https://github.com/3v1n0/authd/blob/115aee8ba68eaef0b2296719956365fda5a44e49/nss/integration-tests/integration_test.go#L146-L146 to make the golden files deterministic:

```suggestion
	user.Name = UserPrefix + "-{{name}}"
```